make-file-list-msys2.sh

```diff
@@ -0,0 +1,128 @@
+#!/bin/sh
+
+die () {
+	echo "$*" >&2
+	exit 1
+}
+
+test -n "$ARCH" &&
+test -n "$BITNESS" ||
+die "Need ARCH and BITNESS to be set"
+
+pacman_list () {
+	package_list=$(for arg
+		do
+			pactree -u "$arg"
+		done |
+		grep -v "^\\($(echo $PACKAGE_EXCLUDES | sed \
+			-e 's/ /\\|/g' \
+			-e 's/mingw-w64-/&\\(i686\\|x86_64\\)-/g')\\)\$" |
+		sort |
+		uniq) &&
+	if test -n "$PACKAGE_VERSIONS_FILE"
+	then
+		pacman -Q $package_list >"$PACKAGE_VERSIONS_FILE"
+	fi &&
+	pacman -Ql $package_list |
+	grep -v '/$' |
+	sed 's/^[^ ]* //'
+}
+
+install_required () {
```

Comment:
I didn't include the stuff about include despite upgrade as guessed it wouldn't be needed